android/app/src/main/java/com/microsoft/reacttestapp/TestApp.kt

```diff
@@ -20,12 +22,18 @@ class TestApp : Application(), HasAndroidInjector, ReactApplication {
     override fun onCreate() {
         super.onCreate()
 
+        val eventConsumers = PackageList(this).packages
+            .filter { it is ReactTestAppLifecycleEvents }
+            .map { it as ReactTestAppLifecycleEvents }
+
+        eventConsumers.forEach { it.onTestAppCreated() }
+
         val testAppComponent = DaggerTestAppComponent.builder()
             .binds(this)
             .build()
-
         testAppComponent.inject(this)
 
+        eventConsumers.forEach { it.onPreInitReactNativeInstance() }
         reactNativeHostInternal.init()
```

Comment:
Is there a post init event as well?

Reply:
I took a look at what iOS does, specifically at `ReactTestAppDidInitializeNotification`. It seems that Android's variant of this event has different semantics, or to be more specific, the event is fired after the application start, not after the RN bridge init. I can introduce a new event on Android that is fired after RN initializes itself, but it will be confusing and inconsistent with iOS.

Ideally, events with similar names should have similar semantics on both platforms. I would repurpose the event on iOS to be sent after application start, and add a new event for post-bridge init. This, however, is going to be a breaking change.

Your call @tido64 and @sweggersen.

Reply:
Not that many people are using these events besides us, I think. Please go ahead and rename them.

Reply:
I like the second suggestion. It's better to do these changes sooner than later, and we need to be consistent between platforms